source/logrepl/copyto_test.go

```diff
@@ -0,0 +1,175 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logrepl
+
+import (
+	"context"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/conduitio/conduit-connector-postgres/test"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+
+	"github.com/jackc/pgx/v4"
+	"github.com/matryer/is"
+)
+
+func TestCopyTo(t *testing.T) {
+	is := is.New(t)
+	ctx := context.Background()
+	conn := test.ConnectSimple(ctx, t, test.RepmgrConnString)
+
+	_, err := conn.Exec(ctx, `create temporary table foo( a int2, b int4, c int8, d varchar, e text, f date, g json)`)
+	is.NoErr(err)
+
+	_, err = conn.Exec(context.Background(), `insert into foo values (0, 1, 2, 'abc	', 'efg', '2000-01-01', '{"abc":"def","foo":"bar"}')`)
+	is.NoErr(err)
+
+	_, err = conn.Exec(context.Background(), `insert into foo values (3, null, null, null, null, null, '{"foo":"bar"}')`)
+	is.NoErr(err)
+
+	w, err := NewCopyDataWriter(ctx, conn, Config{TableName: "foo"})
+	is.NoErr(err)
+
+	go w.Copy(ctx, conn)
+
+	now := time.Now()
+	count := 0
+	for count < 2 {
+		rec, err := w.Next(ctx)
+		is.NoErr(err)
+		is.True(rec.CreatedAt.After(now))
+		count++
+	}
+
+	<-w.done
+	is.NoErr(w.Teardown(ctx))
+}
+
+func TestCopyWriter_Copy(t *testing.T) {
+	is := is.New(t)
+	ctx := context.Background()
+	conn := test.ConnectSimple(ctx, t, test.RepmgrConnString)
+
+	_, err := conn.Exec(ctx, `create temporary table foo( a int2, b int4, c int8, d varchar, e text, f date, g json)`)
+	is.NoErr(err)
+
+	_, err = conn.Exec(context.Background(), `insert into foo values (0, 1, 2, 'abc	', 'efg', '2000-01-01', '{"abc":"def","foo":"bar"}')`)
+	is.NoErr(err)
+
+	_, err = conn.Exec(context.Background(), `insert into foo values (3, null, null, null, null, null, '{"foo":"bar"}')`)
+	is.NoErr(err)
+
+	tx, err := conn.BeginTx(ctx, pgx.TxOptions{
+		IsoLevel:   "REPEATABLE",
+		AccessMode: "READ",
+	})
+	is.NoErr(err)
+	defer is.NoErr(tx.Commit(ctx))
+
+	w, err := NewCopyDataWriter(ctx, tx.Conn(), Config{TableName: "foo"})
+	is.NoErr(err)
+
+	go w.Copy(ctx, conn)
+
+	now := time.Now()
+	count := 0
+	for count < 2 {
+		rec, err := w.Next(ctx)
+		is.NoErr(err)
+		is.True(rec.CreatedAt.After(now))
+		count++
+	}
+
+	<-w.done
+	is.NoErr(w.Teardown(ctx))
+}
+
+func TestCopyDataWriter_Next(t *testing.T) {
+	is := is.New(t)
+	type args struct {
+		config Config
+	}
+	tests := []struct {
+		name       string
+		args       args
+		setupQuery func(conn *pgx.Conn)
+		wantErr    bool
+		want       sdk.Record
+	}{
+		{
+			name: "should parse a copy line into a record",
+			args: args{
+				config: Config{
+					TableName: "foo",
+				},
+			},
+			setupQuery: func(conn *pgx.Conn) {
+				_, err := conn.Exec(context.Background(), `create temporary table foo(a int2, b int4, c int8, d varchar, e text, f date, g json)`)
```

Comment:
I'd love to see a test like the one for [`RelationSet`](https://github.com/ConduitIO/conduit-connector-postgres/blob/db6527de84dd56da9048a265b74f01a6da3de3e4/source/logrepl/internal/relationset_test.go#L43) to be sure we can handle all builtin types.